### PR TITLE
Add text `wrap` and `truncate` props

### PR DESCRIPTION
**Pull request by vladmoroz**
- Add new `wrap` text prop corresponding to the `text-wrap` CSS property
- Add new `truncate` text prop
- Update changelog

### CODE REVIEW

File: packages/radix-ui-themes/changelog.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
     - `gridRow`, `gridRowStart`, `gridRowEnd`
   - Deprecate `shrink` and `grow` props in favour of `flexShrink` and `flexGrow`. The older `shrink` and `grow` props will be removed in the next major release.
   - Update the type signature of the layout props so that code editor suggestions use just space scale values when possible. CSS keywords and other values such as `"auto"` or `"100vw"` are still available as manual string values.
+  - Fix an issue with responsive props when using a breakpoints object without the `initial` key would not apply the default prop value
```
Comment (vladmoroz) [inline packages/radix-ui-themes/changelog.md:24]: Had this fat bug on my list and realized I had accidentally fixed it with that arbitrary props PR

The bug was that components like `<Button size={{ xl: "2" }} />` would appear broken until `xl` and not use the default value

File: packages/radix-ui-themes/src/components/link.props.ts [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
-  weight: typeof textPropDefs.weight;
-  trim: typeof textPropDefs.trim;
+  size: PropDef<(typeof sizes)[number]>;
+  weight: typeof weightProp;
+  trim: typeof trimProp;
+  truncate: typeof truncateProp;
+  wrap: typeof textWrapProp;
   underline: PropDef<(typeof underline)[number]>;
-  color: typeof textPropDefs.color;
-  highContrast: typeof textPropDefs.highContrast;
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/components/link.props.ts:19]: Just making this consistent with how we use exported prop defs everywhere else

File: packages/radix-ui-themes/src/helpers/extract-props.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ function extractProps<
       }
 
       if (propDef.type === 'string') {
-        className = classNames(propDef.className, className);
+        className = classNames(className, propDef.className);
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/helpers/extract-props.ts:50]: Sneaking in a consistency improvement to how prop-driven classnames should all appear at the end of the className list

File: packages/radix-ui-themes/src/styles/utilities/text-wrap.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/*
+ * We are modeling this after the "text-wrap" property, but use "white-space" because it has better support as of 2024.
+ */
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/styles/utilities/text-wrap.css:3]: Everything related to text wrapping in CSS seems like such a mess historically. Apparently, `text-wrap` is supposed to be the spec now, but it's not fully supported yet:
https://caniuse.com/?search=text-wrap

`white-space` can achieve equivalent styles, but handles both whitespace collapsing and text wrapping concerns

File: apps/playground/app/test-text-wrap/page.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                        <Text size="3" as="p">
+                          Under a sky as clear as a sapphire, the audacious wind embarked on yet
+                          another sojourn.
+                        </Text>
+                      </ScrollArea>
+                    </Card>
+
+                    <Card size="3">
+                      <Heading size="3" mb="2">
+                        <Code variant="ghost" color="gray" highContrast>
```
Comment (vladmoroz) [inline apps/playground/app/test-text-wrap/page.tsx:48]: The ghost Code color gotcha [got me again](https://github.com/workos/workos/blob/main/packages/design-system/src/overrides.css#L1-L8). I'll take care of it. @benoitgrelard remind me, did we agree we can do it without a major release? I reckon we talked about it.

File: packages/radix-ui-themes/src/helpers/props/text-wrap.prop.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import type { PropDef } from '..';
+
+const textWrapValues = ['wrap', 'nowrap', 'pretty', 'balance'] as const;
+
+const textWrapProp = {
+  type: 'enum',
+  className: 'rt-r-tw',
+  values: textWrapValues,
+  default: undefined,
```
[the remark this answers is not in the document]
Comment (vladmoroz) [reply packages/radix-ui-themes/src/helpers/props/text-wrap.prop.ts:9]: Good catch, fixed

**Review by benoitgrelard — APPROVED** [review]

**Pull request merged**